### DISCUSSION TITLE
10.4 FoodVision Big Turning Datasets into DataLoaders

### DISCUSSION THREAD

**Discussion by JohnKHancock** [mrdbourke/pytorch-deep-learning#712] [Q&A]
Hello, 

I'm getting an error when I convert the datasets into dataloaders. Specifically, the target labels are not in a tensor. They're integers. When I run the engine.py script, it throws an error when sending the "y" variable to the device. The problem is that the "y" variable or targets are not in a tensor. 

I've reviewed the videos and looked online. What is the best method for converting the target labels from a torchvision.dataset into a tensor?

Thank you for any assistance on this.

![error](https://github.com/mrdbourke/pytorch-deep-learning/assets/22781272/672ae69c-dd03-4bf9-b6d3-c583c681b28e)

`
